decidim-comments/app/controllers/decidim/comments/comments_controller.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Comments
+    # Controller that manages the comments for a commentable object.
+    #
+    class CommentsController < Decidim::Comments::ApplicationController
+      before_action :authenticate_user!, only: [:create]
+      before_action :set_commentable
+      before_action :ensure_commentable!
+
+      helper_method :root_depth, :commentable, :order, :reply?, :reload?
+
+      def index
+        @comments = SortedComments.for(
```

Comment:
Shouldn't we check if comments are enabled for the current commentable?

Reply:
I have added permission checks that ensure this.